centreon-certified/bsm/bsm_connector-apiv2.lua

```diff
@@ -0,0 +1,300 @@
+-- @return the new EventQueue
+--------------------------------------------------------------------------------
+
+function EventQueue.new(params)
+  local self = {}
+  self.fail = false
+
+  local mandatory_parameters = {
+      "http_server_url"
+  }
+
+  -- set up log configuration
+  local logfile = params.logfile or "/var/log/centreon-broker/bsm_connector-apiv2.log"
+  local log_level = params.log_level or 1  
+  
+  -- initiate mandatory objects
+  self.sc_logger = sc_logger.new(logfile, log_level)
+  self.sc_common = sc_common.new(self.sc_logger)
+  self.sc_broker = sc_broker.new(self.sc_logger)
+  self.sc_params = sc_params.new(self.sc_common, self.sc_logger)
+
+  -- checking mandatory parameters and setting a fail flag
+  if not self.sc_params:is_mandatory_config_set(mandatory_parameters, params) then
+    self.fail = true
+  end
+
+  -- overriding default parameters for this stream connector if the default values doesn't suit the basic needs
+  self.sc_params.params.accepted_categories = params.accepted_categories or "neb"
+  self.sc_params.params.accepted_elements = params.accepted_elements or "host_status,service_status"
+  self.sc_params.params.source_ci = params.source_ci or "Centreon"
+  self.sc_params.params.max_output_length = params.max_output_length or 1024
+
+  -- apply users params and check syntax of standard ones
+  self.sc_params:param_override(params)
+  self.sc_params:check_params()
+
+  self.sc_macros = sc_macros.new(self.sc_params.params, self.sc_logger)
+  self.format_template = self.sc_params:load_event_format_file(true)
+  self.sc_params:build_accepted_elements_info()
+  self.sc_flush = sc_flush.new(self.sc_params.params, self.sc_logger)
+
+  local categories = self.sc_params.params.bbdo.categories
+  local elements = self.sc_params.params.bbdo.elements
+
+  self.format_event = {
+    [categories.neb.id] = {
+      [elements.host_status.id] = function () return self:format_event_host() end,
+      [elements.service_status.id] = function () return self:format_event_service() end
+    },
+    [categories.bam.id] = {}
+  }
+  self.send_data_method = {
+    [1] = function (data, element) return self:send_data(data, element) end
+  }
+  
+  -- return EventQueue object
+  setmetatable(self, { __index = EventQueue })
+  return self
+end
+
+-- Format XML file with host infoamtion
+function EventQueue:format_event_host()
+  local xml_host_severity = "<host_severity>" .. self.sc_common:ifnil_or_empty(self.sc_broker:get_severity(self.sc_event.event.host_id) , '0') .. "</host_severity>"
+  local xml_url = self.sc_common:ifnil_or_empty(self.sc_event.event.cache.service.action_url, 'no action url for this host')
+  local xml_notes = "<host_notes>" .. self.sc_common:ifnil_or_empty(self.sc_event.event.cache.service.notes, 'OS not set') .. "</host_notes>"
+  
+  self.sc_event.event.formated_event = {
+    "<event_data>"
+      .. "<hostname>" .. hostname .. "</hostname>"
+      .. xml_host_severity .. xml_notes
```

Comment:
I don't like this piece of code because I'm an annoying person. The thing that I dislike is the concatenation of two variables there in the middle of the xml structure. 

here is your structure: 

<attribute_name> your_var </attribute_name>
<other_attribute> other_var </other_attribute>

and from nowhere it becomes 

<attribute_name> your_var </attribute_name>
<other_attribute> other_var </other_attribute>
a_random_var .. another_random_var
<again_attribute> again_a_var </again_a_var>

So I had to double check what was in those var to check if you were going to break the xml structure. 

In my opinion, you should remove the xml tags from the var and put them in the structure this will look like that 

<hostname> hostname_var </hostname>
<host_severity> xml_host_severity </host_severity>
<xml_notes> xml_notes </xml_notes>

this way, in my opinion, we see the xml tags at a first glance and we do not change the coding style somewhere in the middle of the xml structure